### PR TITLE
[yeb] type-convert checksum and patch values

### PR DESCRIPTION
**Pull request by Caylo**
(to use in yeb, simply replace tuples by lists)

### CODE REVIEW

File: easybuild/framework/easyconfig/types.py [hunk 1]
```diff
@@ -400,6 +400,12 @@ def to_dependencies(dep_list):
     """
     return [to_dependency(dep) for dep in dep_list]
 
+def to_checksums(checksums):
+    if not isinstance(checksums[0], list):
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/types.py]: why the `[0]`?

Comment (Caylo) [inline easybuild/framework/easyconfig/types.py]: checksums can be a list of lists, see https://github.com/hpcugent/easybuild-framework/blob/master/test/framework/easyconfigs/toy-0.0.eb#L10

**Comment by boegel** [comment]
(open/close to trigger Travis)

File: easybuild/framework/easyconfig/types.py [hunk 2]
```diff
@@ -433,16 +439,20 @@ def to_dependencies(dep_list):
     'opt_keys': [],
     'req_keys': ['files', 'dirs'],
 }))
-CHECKABLE_TYPES = [DEPENDENCIES, DEPENDENCY_DICT, NAME_VERSION_DICT, SANITY_CHECK_PATHS_DICT, STRING_OR_TUPLE_LIST,
-                   TUPLE_OF_STRINGS]
+CHECKSUMS = (list, as_hashable({'elem_types': [STRING_OR_TUPLE_LIST]}))
+
+CHECKABLE_TYPES = [CHECKSUMS, DEPENDENCIES, DEPENDENCY_DICT, NAME_VERSION_DICT, SANITY_CHECK_PATHS_DICT,
+                    STRING_OR_TUPLE_LIST, TUPLE_OF_STRINGS]
```
Comment (boegel) [inline easybuild/framework/easyconfig/types.py]: style nit-picking: fix indent to align with line above

File: easybuild/framework/easyconfig/types.py (hunk 1 above)
Comment (boegel) [inline easybuild/framework/easyconfig/types.py]: I'm not sure why this is needed either, it looks wrong to only consider the 1st element.

File: easybuild/framework/easyconfig/types.py [hunk 3]
```diff
@@ -400,6 +400,12 @@ def to_dependencies(dep_list):
     """
     return [to_dependency(dep) for dep in dep_list]
 
+def to_checksums(checksums):
+    if not any(isinstance(checksum, list) for checksum in checksums):
+        return to_list_of_strings_and_tuples(checksums)
+    else:
+        return [to_list_of_strings_and_tuples(cs) for cs in checksums]
```
Comment (boegel) [inline easybuild/framework/easyconfig/types.py]: positive logic (and a docstring) makes this easier to comprehend:

```python
def to_checksums(checksums):
    """Ensure correct element types for list of checksums: convert each element to a tuple."""
    if all(isinstance(cs, basestring) for cs in checksums):
        # if all elements are strings, we convert each individual element to a tuple
        res = [to_list_of_strings_and_tuples(cs) for cs in checksums]
    else:
        res = to_list_of_strings_and_tuples(checksums)

    return res
```

(do check my comment/docstring, I may confuse between list/tuple...)

**Comment by boegel** [comment]
Going in, thanks @Caylo!

**Pull request merged**